runtime/assembly/models/inherentData.ts

```diff
@@ -0,0 +1,80 @@
+import { CompactInt, Bytes, ScaleString, ByteArray } from 'as-scale-codec';
+import { Constants } from '../constants';
+import { DecodedData } from '../codec/decoded-data';
+
+/**
+ * Class representing InherentData transactions in Substrate runtime
+ */
+
+ export class InherentData {
+    /**
+     * A hashtable (Map, in our case) representing the totality of 
+     * inherent extrinsics included in each block. 
+     */
+    public data: Map<ScaleString, ByteArray>;
```

Comment:
As we discussed it would be better to use the native `string` instead of the `scale` one